homework_2/three_words.py

```diff
@@ -0,0 +1,18 @@
+# You are given a string with words and numbers separated by whitespaces (one space).
+# The words contains only letters. You should check if the string contains three words in succession.
+# For example, the string "start 5 one two three 7 end" contains three words in succession.
+
+
+def three_words(txt):
+    words = txt.split()
+    count = 0
+
+    for word in words:
+        if word.isalpha():
+            count += 1
+            if count == 3:
+                return True
+        else:
+            count = 0
+
+    return False
```

Comment:
create one more solution using slicing - sliding window approach:

```
for i in range(len(words) - 2):
    three_words = words[i: i + 3]

```

Reply:
another possible solution is to create 3 slices and use zip:

```
slice_1 = words[::3]
slice_2 = words[1::3]
slice_3 = words[2::3]

for three_words in zip(slice_1, slice_2, slice_3):
    ...

```

Reply:
added sliding window approach solution